src/main/java/com/geekbeast/rhizome/pods/hazelcast/BaseHazelcastInstanceConfigurationPod.java

```diff
@@ -46,25 +48,46 @@ public Config getHazelcastConfiguration() {
         return config;
     }
 
-    protected NetworkConfig getNetworkConfig( HazelcastConfiguration hzConfiguration ) {
+    @Bean
```

Comment:
`@Inject` declares a dependency on `ClientConfig` for the `HazelcastPod`. This means that a client configuration must be present for the server to start, even if its intended to be run in server mode. It sounds like we need a class with two `Optional<...>` parameters for configuration. That class should be the bean exposed to the HazelcastPod. It should fail if there is not at least one of the necessary configuration classes provided.